src/python/test/test_remote_datasets.py

```diff
@@ -0,0 +1,45 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Hind Montassif
+#
+# Copyright (C) 2021 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+
+from gudhi.datasets import remote
+import re
+
+def test_fetch_remote_datasets():
+    # Test files download from given urls
+    path_file_dw = remote.fetch("https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/spiral_2d.csv", "spiral_2d.csv")
+    names_dw = re.split(r' |/|\\', path_file_dw)
+    assert 'remote_datasets' == names_dw[0]
+    assert 'spiral_2d.csv' == names_dw[1]
+
+    path_file_dw = remote.fetch("https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/sphere3D_pts_on_grid.off", "sphere3D_pts_on_grid.off")
+    names_dw = re.split(r' |/|\\', path_file_dw)
+    assert 'remote_datasets' == names_dw[0]
+    assert 'sphere3D_pts_on_grid.off' == names_dw[1]
+
+
+    # Test files download with checksums provided
+    path_file_dw = remote.fetch("https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/spiral_2d.csv", "spiral_2d.csv",
+                                file_checksum = '37530355d980d957c4ec06b18c775f90a91e446107d06c6201c9b4000b077f38')
```

Comment:
If file is already downloaded, the checksum is no more checked. Please test before with the checksum, then without it.

A first test with a wrong checksum that triggers an exception would be a plus.
```python
with pytest.raises(OSError):
    path_file_dw = remote.fetch("https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/spiral_2d.csv", "spiral_2d.csv", file_checksum = 'XXXXXXXXXX')
```

Reply:
It depends if fetch is made private or not; if public and checksum is not provided I would say that verifying the checksum of the found local file would be good if the user provides it for a second (or more) fetch call. Or, we can also make the checksum mandatory ... In that case, it would be enough to verify it only the first time (unless the user downloads the file manually (or by other means other than fetch) and the download fails without him noticing and then uses fetch another time).

Reply:
Done.